Source/SwiftLintFramework/Rules/Lint/DuplicateConditionsRule.swift

```diff
@@ -0,0 +1,227 @@
+import SwiftSyntax
+
+struct DuplicateConditionsRule: SwiftSyntaxRule, ConfigurationProviderRule {
+    var configuration = SeverityConfiguration(.warning)
+
+    init() {}
+
+    static let description = RuleDescription(
+        identifier: "duplicate_conditions",
+        name: "Duplicate Conditions",
+        description: "Duplicate sets of conditions in the same branch instruction should be avoided",
+        kind: .lint,
+        nonTriggeringExamples: [
+            Example("""
+                if x < 5 {
+                  foo()
+                } else if y == "s" {
+                  bar()
+                }
+            """),
+            Example("""
+                if x < 5 {
+                  foo()
+                }
+                if x < 5 {
+                  bar()
+                }
+            """),
+            Example("""
+                if x < 5, y == "s" {
+                  foo()
+                } else if x < 5 {
+                  bar()
+                }
+            """),
+            Example("""
+                switch x {
+                case \"a\":
+                  foo()
+                  bar()
+                }
+            """),
+            Example("""
+                switch x {
+                case \"a\" where y == "s":
+                  foo()
+                case \"a\" where y == "t":
+                  bar()
+                }
+            """),
+            Example("""
+                if let x = maybeAbc {
+                  foo()
+                } else if let x = maybePqr {
+                  bar()
+                }
+            """),
+            Example("""
+                if let x = maybeAbc, let z = x.maybeY {
+                  foo()
+                } else if let x = maybePqr, let z = x.maybeY {
+                  bar()
+                }
+            """),
+            Example("""
+                if case .p = x {
+                  foo()
+                } else if case .q = x {
+                  bar()
+                }
+            """)
+        ],
```

Comment:
What do you think about adding a non-triggering example of two nested if-statements with the same condition?

```swift
if true {
    if true { foo() }
}
```